tutorials/cifar10-distributed.ipynb

```diff
@@ -0,0 +1,1319 @@
+        "        criterion,\n",
+        "        device=device,\n",
+        "        non_blocking=True,\n",
+        "        output_transform=lambda x, y, y_pred, loss: {\"batch loss\": loss.item()},\n",
+        "        amp_mode=amp_mode,\n",
+        "        scaler=scaler,\n",
+        "    )\n",
+        "    trainer.logger = logger\n",
+        "\n",
+        "    to_save = {\n",
+        "        \"trainer\": trainer,\n",
+        "        \"model\": model,\n",
+        "        \"optimizer\": optimizer,\n",
+        "        \"lr_scheduler\": lr_scheduler,\n",
+        "    }\n",
+        "    metric_names = [\n",
+        "        \"batch loss\",\n",
+        "    ]\n",
+        "\n",
+        "    common.setup_common_training_handlers(\n",
+        "        trainer=trainer,\n",
+        "        train_sampler=train_sampler,\n",
+        "        to_save=to_save,\n",
+        "        save_every_iters=config[\"checkpoint_every\"],\n",
+        "        save_handler=get_save_handler(config),\n",
+        "        lr_scheduler=lr_scheduler,\n",
+        "        output_names=metric_names if config[\"log_every_iters\"] > 0 else None,\n",
+        "        with_pbars=False,\n",
+        "        clear_cuda_cache=False,\n",
+        "    )\n",
+        "\n",
+        "    if config[\"resume_from\"] is not None:\n",
+        "        checkpoint = load_checkpoint(config[\"resume_from\"])\n",
+        "        Checkpoint.load_objects(to_load=to_save, checkpoint=checkpoint)\n",
+        "\n",
+        "    return trainer"
+      ],
+      "execution_count": 18,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "U5ius6EM9aiG"
+      },
+      "source": [
+        "## Evaluator\n",
+        "\n",
+        "The evaluator will be created via [`create_supervised_evaluator()`](https://pytorch.org/ignite/generated/ignite.engine.create_supervised_evaluator.html#ignite.engine.create_supervised_evaluator) which internally will:\n",
+        "1. Set the `model` to `eval()` mode.\n",
+        "2. Move the batch to `device` used in current distributed configuration.\n",
+        "3. Perform forward pass. If AMP is enabled, `autocast` will be on.\n",
+        "4. Store the predictions and labels in `state.output` to compute metrics.\n",
+        "\n",
+        "Finally, we will attach relevant Ignite metrics to the `evaluator`. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "JPVzU6CqNlE4"
+      },
+      "source": [
+        "def create_evaluator(model, metrics, config):\n",
+        "    device = idist.device()\n",
+        "\n",
+        "    amp_mode = \"amp\" if config[\"with_amp\"] else None\n",
+        "    evaluator = create_supervised_evaluator(\n",
+        "        model, metrics=metrics, device=device, non_blocking=True, amp_mode=amp_mode\n",
+        "    )\n",
+        "\n",
+        "    for name, metric in metrics.items():\n",
```

Comment:
This is already done in `create_supervised_evaluator`. 

https://github.com/pytorch/ignite/blob/f9cd677de7aa87e9ecb9ad658c7170dddb85f444/ignite/engine/__init__.py#L579